spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporterFactory.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.spark.metrics.sink.statsd;
+
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricAttribute;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+
+import io.dropwizard.metrics.BaseReporterFactory;
+
+public class StatsdReporterFactory extends BaseReporterFactory {
```

Comment:
do we really need to extend `BaseReporterFactory` to build `MetricFilter`? I'd vote for manual filter construction in order to get rid of `dropwizard` dependency and depend only on `codahale.metrics` - this way we can avoid shading and after bumping Spark own codahale dependency can safely rely on `spark.core` with provided scope.

Reply:
Added `addIfMatches` method.